testsuite/Makefile

```diff
@@ -17,46 +17,43 @@ NO_PRINT=`$(MAKE) empty --no-print-directory >/dev/null 2>&1 \
 FIND=find
 include ../config/Makefile
 
+# Maximum number of retries for directories with failing tests
+# Default = 1 for native Windows, 0 for all other platforms
+ifeq ($(filter msvc mingw,$(TOOLCHAIN)),)
+MAX_RETRIES=0
+else
+MAX_RETRIES=1
+endif
+
 .PHONY: default
 default:
 	@echo "Available targets:"
 	@echo "  all             launch all tests"
+	@echo "  all-foo         launch all tests beginning with foo"
 	@echo "  list FILE=f     launch the tests listed in f (one per line)"
 	@echo "  one DIR=p       launch the tests located in path p"
 	@echo "  promote DIR=p   promote the reference files for the tests in p"
 	@echo "  lib             build library modules"
 	@echo "  clean           delete generated files"
 	@echo "  report          print the report for the last execution"
+	@echo
+	@echo "all* and list can automatically re-run failed test directories if MAX_RETRIES"
+	@echo "permits (default value = $(MAX_RETRIES))"
 
 .PHONY: all
 all: lib
 	@for dir in tests/*; do \
 	  $(MAKE) $(NO_PRINT) exec-one DIR=$$dir; \
 	done 2>&1 | tee _log
+	@$(MAKE) $(NO_PRINT) retries
 	@$(MAKE) report
 
-all-basic: lib
-	@for dir in tests/basic*; do \
-	  $(MAKE) $(NO_PRINT) exec-one DIR=$$dir; \
-	done 2>&1 | tee _log
-	@$(MAKE) report
-
-all-lib: lib
-	@for dir in tests/lib-*; do \
-	  $(MAKE) $(NO_PRINT) exec-one DIR=$$dir; \
-	done 2>&1 | tee _log
-	@$(MAKE) report
-
-all-typing: lib
-	@for dir in tests/typing-*; do \
-	  $(MAKE) $(NO_PRINT) exec-one DIR=$$dir; \
-	done 2>&1 | tee _log
-	@$(MAKE) report
-
-all-tool: lib
-	@for dir in tests/tool-*; do \
+.PHONY: all-%
+all-%: lib
```

Comment:
It's not precisely identical to the previous one (needing all-basic to include tests/basic means that all-tool would include tests/tool if it existed, etc.) but I figured that was OK!